storage/operation/stats.go

```diff
@@ -0,0 +1,168 @@
+package operation
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module/util"
+	"github.com/onflow/flow-go/storage"
+)
+
+// Stats holds statistics for a single prefix group.
+type Stats struct {
+	Count       int
+	TotalSize   int
+	MinSize     int
+	MaxSize     int
+	AverageSize float64
+}
+
+// SummarizeKeysByFirstByteConcurrent iterates over all prefixes [0x00..0xFF] in parallel
+// using nWorker goroutines. Each worker handles one prefix at a time until all are processed.
+//
+// The storage.Reader must be able to create multiple iterators concurrently.
+func SummarizeKeysByFirstByteConcurrent(log zerolog.Logger, r storage.Reader, nWorker int) (map[byte]Stats, error) {
+	// We'll have at most 256 possible prefixes (0x00..0xFF).
+	// Create tasks (one per prefix), a results channel, and a wait group.
+	taskChan := make(chan byte, 256)
+	resultChan := make(chan struct {
+		prefix byte
+		stats  Stats
+		err    error
+	}, 256)
+
+	var wg sync.WaitGroup
+
+	// Start nWorker goroutines.
+	for i := 0; i < nWorker; i++ {
+		wg.Add(1)
+		go func() {
```

Comment:
since the function can return an error, consider using [errorgroup.GO](https://pkg.go.dev/golang.org/x/sync/errgroup#Group.Go).